oshi-core/src/main/java/oshi/software/os/unix/solaris/SolarisOSProcess.java

```diff
@@ -281,4 +281,76 @@ private boolean updateAttributes(String[] split) {
 
         return true;
     }
+
+    /**
+     * Merges results of a ps and prstat query, since Solaris thread details are not
+     * available in a single command. Package private to permit access by
+     * SolarisOSThread.
+     *
+     * @param threadListInfo1
+     *            output from ps command.
+     * @param threadListInfo2
+     *            output from the prstat command.
+     * @return a map with key as thread id and an array of command outputs as value
+     */
+    static Map<Integer, String[]> parseAndMergeThreadInfo(List<String> threadListInfo1,
+            List<String> threadListInfo2) {
+        Map<Integer, String[]> map = new HashMap<>();
+        final String[] split = new String[9];
+        // 0-lwpid, 1-state,2-elapsedtime,3-kerneltime, 4-usertime, 5-address,
+        // 6-priority
+        if (threadListInfo1.size() > 1) { // first row is header
+            threadListInfo1.stream().skip(1).forEach(threadInfoPs -> {
+                String[] splitPs = ParseUtil.whitespaces.split(threadInfoPs.trim(), 6);
+                IntStream.rangeClosed(0, 6).forEach(idx -> split[idx] = splitPs[idx]);
+                map.put(ParseUtil.parseIntOrDefault(splitPs[0], 0), split);
+            });
+            // 0-pid, 1-username, 2-usertime, 3-sys, 4-trp, 5-tfl, 6-dfl, 7-lck, 8-slp,
+            // 9-lat, 10-vcx, 11-icx, 12-scl, 13-sig, 14-processid/lwpid
+            if (threadListInfo2.size() > 1) { // first row is header
+                threadListInfo2.stream().skip(1).forEach(threadInfo -> {
+                    String[] splitPrstat = ParseUtil.whitespaces.split(threadInfo.trim(), 10);
+                    String[] processLwpid = ParseUtil.slash.split(splitPrstat[14].trim()); // format is process/lwpid
```

Comment:
ok

Reply:
This looks good. One more thing I'm adding (not because it will ever happen but because I code very defensively!) is in the conditional on line 320, to also check that `idxForwardSlash` is less than length (to avoid using substring out of range) or less than length-1 (to ensure that we actually have something to parse later).